src/transformers/modeling_t5.py

```diff
@@ -949,10 +976,8 @@ def forward(
 
         decoder_outputs = (lm_logits,) + decoder_outputs[1:]  # Add hidden states and attention if they are here
         if lm_labels is not None:
-            shift_logits = lm_logits[..., :-1, :].contiguous()
-            shift_labels = lm_labels[..., 1:].contiguous()
```

Comment:
I don't think this should be done in T5. Since the `decoder_input_ids` should always start with the `pad_token_id` and the `lm_labels` always correspond to the `decoder_input_ids` shifted to the left by one, we should not shift here again, I think @thomwolf